### PR TITLE
Geant4 initialisation clean-up for DD4Hep migration

### PR DESCRIPTION
**Pull request by civanch**
#### PR description:
Initialisations of Geant4 geometry, regions, and production cuts are concentrated in DDDWorld class from SimG4Core/Geometry sub-package. Initialisation of magnetic field is performed together with SensitiveDetector definition only inside RunManagerMTWorker class. This modifications make transition to DD4Hep more transparent.

Correctness of this PR means identical results.

#### PR validation:
private

#### if this PR is a backport please specify the original PR:

Before submitting your pull requests, make sure you followed this checklist:
- verify that the PR is really intended for the chosen branch
- verify that changes follow [CMS Naming, Coding, And Style Rules](http://cms-sw.github.io/cms_coding_rules.html)
- verify that the PR passes the basic test procedure suggested in the [CMSSW PR instructions](https://cms-sw.github.io/PRWorkflow.html)

### CODE REVIEW

**Comment by civanch** [comment]
please test

**Comment by civanch** [comment]
please test

**Comment by civanch** [comment]
+1

File: SimG4Core/Geometry/interface/DDDWorld.h [hunk 1]
```diff
@@ -4,19 +4,22 @@
...
 namespace cms {
-  class DDDetector;
-}  // namespace cms
+  class DDCompactView;
+}
 
 class DDDWorld {
 public:
+  DDDWorld(const DDCompactView *pDD,
+           const cms::DDCompactView *pDD4hep,
```
Comment (fabiocos) [inline SimG4Core/Geometry/interface/DDDWorld.h]: @civanch so for the time being you are always assuming both geometries in the interface, just one of the two will be a ```nullptr``` , but the interface is now universal and unique, right?

Comment (civanch) [reply SimG4Core/Geometry/interface/DDDWorld.h]: @fabiocos , yes, exactly.

**Comment by fabiocos** [comment]
+1

**Pull request merged**